### ISSUE TITLE
ObjectStore Python Support: Extend Functionalities From C# 

### ISSUE THREAD

**Issue by AlexCatarino** [QuantConnect/Lean#3963]
#### Expected Behavior
Be able to use helper methods available in C# that simplify saving and reading process.  

#### Actual Behavior
`ReadJson`, `SaveJson`, `ReadXml` and `SaveXml` does not support list types:
```python
values = self.ObjectStore.ReadJson[IndicatorDataPoint[]](self.SPY_Close_ObjectStore_Key)
``` 
#### Reproducing the Problem
Runs `ObjectStoreExampleAlgorithm.py` with the code snippet above.

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

**Comment by jaredbroad** [comment]
Closed in #3956

**Comment by jaredbroad** [comment]
Note for posterity here: the specific example `ReadJson[IndicatorDataPoint[]]` is not working due to  pythonnet; but ReadJson[IndicatorDataPoint] works just fine. So the casting of array object types is a pythonnet issue not a LEAN one as such.
